### DISCUSSION TITLE
Can mpv parse php as input?

### DISCUSSION THREAD

**Discussion by wbtcpip2** [mpv-player/mpv#14789] [Q&A] [answered]
with VLC i can open this stream:
https://www.radiosalento.net/filmdoc/filmrandom.php
(that return an mp4 file "https://radiosalento.net/filmdoc/i 12 Apostoli di Gesu` Cristo - Ecco come Morirono.mp4") 

while i cannot with mpv 
Failed to recognize file format.
Exiting... (Errors when loading file)

is there any setting that allow mpv to parse a php as input?

**Comment by kasper93** [comment]
`mpv "$(curl https://www.radiosalento.net/filmdoc/filmrandom.php)"`

**Comment by wbtcpip2** [comment]
`mpv "$(curl https://www.radiosalento.net/filmdoc/filmrandom.php)"
[file] Cannot open file '$(curl https://www.radiosalento.net/filmdoc/filmrandom.php)': Invalid argument
Failed to open $(curl https://www.radiosalento.net/filmdoc/filmrandom.php).
Exiting... (Errors when loading file)`

is this also valid on windows?

**Reply by kasper93** [reply]
Yes, but only from powershell.

**Reply by verygoodlee** [reply]
solution for cmd.exe
```cmd
for /f "delims=" %a in ('curl --silent "https://www.radiosalento.net/filmdoc/filmrandom.php"') do mpv "%a"
```

**Comment by wbtcpip2** [comment, marked as answer]
ok as i use libmpv i will parse the uri before passing it to MPVCommand loadfile

i was wondering if mpv had an itnernal parser.

thank you

**Reply by kasper93** [reply]
No, mpv itself will not resolve this indirection. You can extract url before passing to mpv or add extractor to ytdl so it gets done internally.
